snap/snapcraft.yaml

```diff
@@ -71,8 +71,33 @@ grade: stable
 confinement: classic
 
 parts:
+  # The cryptography library (indirect dependency via craft-store) requires rust, but only ships wheels for amd64 and arm64.
+  # As of cryptography 39, it requires rust v1.48.0 or newer, which is available from focal-updates.
+  # This part contains the full set of build dependencies for cryptography.
+  # NOTE: cryptography requires at least pip version 22 for building.
+  cryptography-deps:
+    plugin: nil
+    build-packages:
+      - cargo
+      - rustc
+      - pkg-config
+      - python3.8-dev
+      - libpython3.8-dev
+      - libssl-dev
+      - libffi-dev
+      - python3-wheel
+      - python3-pip
+      - python3-setuptools-scm
+    override-build: |
+      pip install 'pip>=22.0' 'setuptools>=61.0.0'
+      # Dirty workaround - builds the cryptography wheel for later installation.
+      # This must be exactly the version referenced in requirements.txt, so this is fragile.
+      pip wheel cryptography==40.0.2
```

Comment:
how about (but in the charmcraft part, with the other suggestions)

```
parts:
  charmcraft:
    ..
    python-packages: [wheel, setuptools>=61.0.0, pip>=22.0]
    override-build: |
      pip wheel `$(grep` ^cryptography= requirements.txt)
      snapcraftctl build
```

This might work and be more straightforward

Reply:
Whenever I try to build the cryptography wheel from another part it fails. At first I thought this was because `python-packages` was only getting installed during the build step, but even installing setuptools and pip beforehand was messing with it.

Reply:
ah, it might be the python plugin installing it's dependencies

Reply:
I riffed on what you did and came up with something much better than what I'd had. 

I've also introduced a workflow test for remote builds, but commented it out until we can get launchpad credentials as appropriate for the build.